### PR TITLE
Factions: Fix crash & Performance Improvements

### PR DESCRIPTION
**Pull request by DeathlyPlays**

### CODE REVIEW

File: wavelength-plugins/factions.js [hunk 1]
```diff
@@ -708,7 +708,7 @@ exports.commands = {
 			factions[factionid].ranks[toId(rank)].users.splice(factions[factionid].ranks[toId(rank)].users.indexOf(toId(targetUser)), 1);
 			write();
 			if (Users(targetUser) && Users(targetUser).connected) {
-				Users(targetUser).send("|popup||html|" + user.name + " has removed you from the faction rank " + Chat.escapeHTML(rank) + " in " +
+				Users(targetUser).send("|popup||html|" + WL.nameColor(user.name) + " has removed you from the faction rank " + Chat.escapeHTML(rank) + " in " +
```
Comment (HoeenCoder) [inline wavelength-plugins/factions.js]: Usernames should be HTML escaped in HTML outputs.

File: wavelength-plugins/factions.js [hunk 2]
```diff
@@ -673,7 +673,7 @@ exports.commands = {
 			factions[factionid].ranks[toId(rank)].users.push(targetUser.userid);
 			write();
 			rank = factions[factionid].ranks[toId(rank)].title;
-			targetUser.send("|popup||html|" + user.name + " has set your faction rank in " + Chat.escapeHTML(factions[factionid].name) + " to " + Chat.escapeHTML(rank) + ".");
+			targetUser.send("|popup||html|" + WL.nameColor(user.name) + " has set your faction rank in " + Chat.escapeHTML(factions[factionid].name) + " to " + Chat.escapeHTML(rank) + ".");
```
Comment (HoeenCoder) [inline wavelength-plugins/factions.js]: Usernames should be HTML escaped in HTML outputs.

Comment (DeathlyPlays) [reply wavelength-plugins/factions.js]: Wait WL.nameColor = html?

Comment (DeathlyPlays) [reply wavelength-plugins/factions.js]: So why "escape"

Comment (HoeenCoder) [reply wavelength-plugins/factions.js]: `Wl.nameColor(Chat.escapeHTML(user.name))`

Comment (HoeenCoder) [reply wavelength-plugins/factions.js]: This is so that usernames with HTML charachters dont cause issues

Comment (DeathlyPlays) [reply wavelength-plugins/factions.js]: Yea

**Pull request merged**